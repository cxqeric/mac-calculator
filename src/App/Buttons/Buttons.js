import React from 'react';
import PropTypes from 'prop-types';

import Calculator from '../Calculator/Calculator';

class Buttons extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ])
  };

  static defaultProps = {
    children: null
  };

  static contextTypes = {
    [Calculator.CALCULATOR_CONTEXT]: PropTypes.shape({
      initial: PropTypes.bool
    })
  };

  render() {
    return (
      <div className="buttons">
        {this.props.children}
      </div>
    );
  }
}

export default Buttons;

