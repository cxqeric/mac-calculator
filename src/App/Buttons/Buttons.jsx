import React from 'react';
import PropTypes from 'prop-types';

const Buttons = ({ children }) => (
  <div className="buttons">
    {children}
  </div>
);

Buttons.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Buttons.defaultProps = {
  children: null
};


export default Buttons;

