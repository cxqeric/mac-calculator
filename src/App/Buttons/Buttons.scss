.buttons {
  flex-basis: 80%;
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}
