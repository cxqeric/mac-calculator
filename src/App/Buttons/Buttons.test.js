import React from 'react';
import { shallow } from 'enzyme';

import Buttons from './Buttons';


describe('Buttons', () => {
  test('renders empty buttons', () => {
    const wrapper = shallow(<Buttons />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders children', () => {
    const wrapper = shallow(<Buttons> <div className="child">Child Div</div> </Buttons>);
    expect(wrapper).toMatchSnapshot();
  });
});
