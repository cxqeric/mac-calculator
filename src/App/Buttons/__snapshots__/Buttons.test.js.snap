// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders children 1`] = `
<div
  className="buttons"
>
   
  <div
    className="child"
  >
    Child Div
  </div>
   
</div>
`;

exports[`Buttons renders empty buttons 1`] = `
<div
  className="buttons"
/>
`;
